### PR TITLE
{vis}[GCCcore/12.3.0] gnuplot v5.4.8

### PR DESCRIPTION
**Pull request by stefan-wolfsheimer**
(created using `eb --new-pr`)

edit: requires:
* #18558

### CODE REVIEW

**Comment by boegel** [comment]
@stefan-wolfsheimer missing easyconfig for `wxWidgets/3.2.2.1-GCCcore-12.3.0`

File: easybuild/easyconfigs/g/gnuplot/gnuplot-5.4.8-GCCcore-12.3.0.eb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    ('cairo', '1.17.8'),
+    ('libjpeg-turbo', '2.1.5.1'),
+    ('libpng', '1.6.39'),
+    ('libgd', '2.3.3'),
+    ('Pango', '1.50.14'),
+    ('libcerf', '2.3'),
+    ('X11', '20230603'),
+    ('Qt5', '5.15.10'),
+    ('Lua', '5.4.6'),
+    ('wxWidgets', '3.2.2.1')
```
Comment (Micket) [inline easybuild/easyconfigs/g/gnuplot/gnuplot-5.4.8-GCCcore-12.3.0.eb:32]: It really needs Qt *and* wxWidgets??

Comment (Micket) [reply easybuild/easyconfigs/g/gnuplot/gnuplot-5.4.8-GCCcore-12.3.0.eb:32]: Oh it was introduced here:
https://github.com/easybuilders/easybuild-easyconfigs/pull/17483#issuecomment-1463816755

I disagree with the conclusion though, i think it means we should add `--without-wx` (which i think is the right flag)

(also, why was gnuplot-5.4.6 moved to GCC level randomly?)

Comment (stefan-wolfsheimer) [reply easybuild/easyconfigs/g/gnuplot/gnuplot-5.4.8-GCCcore-12.3.0.eb:32]: @Micket I removed the wxWidgets dependency and changed the toolchain to GCC for consistency.

Comment (Micket) [reply easybuild/easyconfigs/g/gnuplot/gnuplot-5.4.8-GCCcore-12.3.0.eb:32]: I think moving to GCC was a mistake. I actually want to see it moved back to GCCcore
https://github.com/easybuilders/easybuild-easyconfigs/pull/18648

Comment (Micket) [reply easybuild/easyconfigs/g/gnuplot/gnuplot-5.4.8-GCCcore-12.3.0.eb]: You actually added both variants. When updating an easyconfig with the command line, you can use `:gnuplot-5.4.8-GCC-12.3.0.eb` to indicate that you want to remove it.

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by Micket** [comment]
@boegel wait what? are we dropping the GCCcore version? Why?

**Comment by boegel** [comment]
> @boegel wait what? are we dropping the GCCcore version? Why?

Oh, I thought there was a reason that we must use `GCC`... But that was of `wxWidgets` I guess, and we should go with `GCCcore` instead?

**Comment by boegel** [comment]
@boegelbot please test @ jsc-zen2

**Comment by boegel** [comment]
@Micket OK now? Sorry for the mess/confusion :)

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3103.skitty.os - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/boegel/c8c61cefe9b0a0cbfa1e3c5b15ac0a36 for a full test report.

**Comment by Micket** [comment]
Weeeeeell I was also hoping to maybe try this with Qt6 (but i'm not even sure if it's supported at all)

**Comment by branfosj** [comment]
> Weeeeeell I was also hoping to maybe try this with Qt6 (but i'm not even sure if it's supported at all)

It does not look like it.

```
$ ./configure | grep qt
  --with-qt =qt4 =qt5 =no  Qt terminal (default autodetect)
```

And if I try a `--with-qt=qt6` the configure log has
```
checking for QT... configure: WARNING:
Package requirements (Qt5Core Qt5Gui Qt5Network Qt5Svg Qt5PrintSupport) were not met:

Package 'Qt5Core', required by 'virtual:world', not found
Package 'Qt5Gui', required by 'virtual:world', not found
Package 'Qt5Network', required by 'virtual:world', not found
Package 'Qt5Svg', required by 'virtual:world', not found
Package 'Qt5PrintSupport', required by 'virtual:world', not found

Consider adjusting the PKG_CONFIG_PATH environment variable if you
installed software in a non-standard prefix.

Alternatively, you may set the environment variables QT_CFLAGS
and QT_LIBS to avoid the need to call pkg-config.
See the pkg-config man page for more details.

checking for QT... configure: WARNING:
Package requirements (QtCore >= 4.5 QtGui >= 4.5 QtNetwork >= 4.5 QtSvg >= 4.5) were not met:

Package 'QtCore', required by 'virtual:world', not found
Package 'QtGui', required by 'virtual:world', not found
Package 'QtNetwork', required by 'virtual:world', not found
Package 'QtSvg', required by 'virtual:world', not found

Consider adjusting the PKG_CONFIG_PATH environment variable if you
installed software in a non-standard prefix.

Alternatively, you may set the environment variables QT_CFLAGS
and QT_LIBS to avoid the need to call pkg-config.
See the pkg-config man page for more details.

The Qt terminal will not be compiled.
```

It looks like it will be supported in the future (in version 6.x ?): https://github.com/gnuplot/gnuplot/commit/e458b88e481e8af4ef2d97c9af79fdf40e40ed81

File: easybuild/easyconfigs/g/gnuplot/gnuplot-5.4.8-GCCcore-12.3.0.eb [hunk 2]
```diff
@@ -0,0 +1,45 @@
+easyblock = 'ConfigureMake'
+
+name = 'gnuplot'
+version = '5.4.8'
+
+homepage = 'http://gnuplot.sourceforge.net'
+description = """Portable interactive, function plotting utility"""
+
+toolchain = {'name': 'GCC', 'version': '12.3.0'}
```
Comment (Micket) [inline easybuild/easyconfigs/g/gnuplot/gnuplot-5.4.8-GCCcore-12.3.0.eb:9]:
```suggestion
toolchain = {'name': 'GCCcore', 'version': '12.3.0'}
```

**Comment by Micket** [comment]
@boegelbot please test @ jsc-zen2

**Review by Micket — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Going in, thanks @stefan-wolfsheimer!

**Pull request merged**